### ISSUE TITLE
JSX: Error - Unexpected token <

### ISSUE THREAD

**Issue by fzzzt** [eslint/eslint#2943]
espree 2.0.4
eslint 0.24.0

I am getting this error in SublimeLinter-contrib-eslint. In the console it seems to be running but I'm not 100% sure that my config is being applied and that is resulting in this error, so I'm not sure if it's my linter config, SublimeLinter config, or what. The console logs the following (the [...] are my replacing of long paths):

    SublimeLinter: eslint: MyView.jsx ['[...]\\eslint.cmd', '--format', 'compact', '--stdin', '--stdin-filename', '__RELATIVE_TO_FOLDER__'] 
    SublimeLinter: eslint output: [...]MyView.jsx: line 3, col 11, Error - Unexpected token <

Code:

    var MyView = React.createClass({
    	render: function() {
    		return (<div>Coming Soon!</div>);
    	}
    });

My SublimeLinter configuration:

    "default": {
        [...]
        "linters": {
            "eslint": {
                "env": {
                    "browser": true,
                    "jquery": true
                },
                "ecmaFeatures": {
                    "jsx": true
                }
            }
        },
    }

I guess the first question is: _should_ this `eslint` configuration work, assuming SublimeLinter is working properly? Any other hints as to what might be wrong?

**Comment by ilyavolodin** [comment]
Config looks fine (I think). Can you try running eslint on cli instead and see if the same issue shows up?

**Comment by fzzzt** [comment]
I didn't mention that this is on Windows 8 x64. I'm not sure how to specify JSX on the CLI but running variations of this:

    eslint --global jsx:true --env browser,jquery FulfillmentVendorsView.jsx

Didn't change the outcome, same error. If I create a `.eslintrc` file with these contents:

    {
        "ecmaFeatures": {
            "jsx": true
        },
        "env": {
        	"browser": true,
    	    "jquery": true
        }
    }

It sort of works. I get this output:

    MyView.jsx
       1:4   error  MyView is defined but never used  no-unused-vars
       1:29  error  "React" is not defined                            no-undef
       6:10  error  Missing "use strict" statement                    strict
       8:18  error  Strings must use doublequote                      quotes
       8:40  error  Strings must use doublequote                      quotes
       9:18  error  Strings must use doublequote                      quotes
       9:40  error  Strings must use doublequote                      quotes
       13:9   error  Missing "use strict" statement                    strict

    ✖ 8 problems (8 errors, 0 warnings)

The only real issue here is that I was expecting browser:true and/or jsx:true to not trigger the "React is not defined" error. So I guess there is something wrong with how SublimeLinter is passing things to `eslint`, _and_ something wrong with how I'm running `eslint`, which brings up a couple follow-up questions:

1. Is there a smart way to disable the React error without turning off all no-undef errors?
2. Is there a place I can put this `.eslintrc

**Comment by ilyavolodin** [comment]
1. You need to install [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) to get rid of React error.
2. You can put .eslintrc file into your home directory, if ESLint doesn't find any local .eslintrc file it will look at your home directory for one.

I believe this issue can now be closed. If I'm not right, please feel free to reopen it.

**Comment by fzzzt** [comment]
Great, thank you. In case anyone else happens by with my initial problem: SublimeLinter doesn't (currently) pass the configuration along, so I ended up creating a `.eslintrc` file in my Documents folder and passed a command-line argument to `eslint` via this SublimeLinter configuration:

        "linters": {
            "eslint": {
                "args": "--config C:/Users/.../Documents/.eslintrc"
            }
        },

I tried eslint-plugin-react and it causes eslint to crash; I guess I'll head over there next.

Thanks again for the help.
